src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs

```diff
@@ -162,56 +115,35 @@ public static void MoveFile(string sourceFullPath, string destFullPath, bool ove
         {
             // The desired behavior for Move(source, dest) is to not overwrite the destination file
             // if it exists. Since rename(source, dest) will replace the file at 'dest' if it exists,
-            // link/unlink are used instead. Rename is more efficient than link/unlink on file systems
-            // where hard links are not supported (such as FAT). Therefore, given that source file exists,
-            // rename is used in 2 cases: when dest file does not exist or when source path and dest
-            // path refer to the same file (on the same device). This is important for case-insensitive
+            // link/unlink are used instead. rename is used when the source path and dest path refer
```

Comment:
comment about performance of trying link() on FAT is clearly not correct because the syscall fast-fails so if you have to perform any checks before deciding to rename() it's slower.